app/src/features/apiClient/screens/apiClient/components/clientView/APIClientView.tsx

```diff
@@ -71,6 +76,54 @@ const APIClientView: React.FC<Props> = ({ apiEntry, notifyApiRequestFinished })
     };
   }, [apiEntry]);
 
+  const parseRequestData = useCallback(
+    (request: RQAPI.Request): RQAPI.Request => {
+      const sanitizedQueryParams = removeEmptyKeys(request.queryParams);
+      const sanitizedHeaders = removeEmptyKeys(request.headers);
+
+      const parsedRequest: RQAPI.Request = {
+        ...request,
+        url: renderVariables(request.url),
+      };
+
+      const parsedQueryParams = sanitizedQueryParams.map((queryParamKeyValue) => {
+        return {
+          id: queryParamKeyValue.id,
+          key: renderVariables(queryParamKeyValue.key),
+          value: renderVariables(queryParamKeyValue.value),
+        };
+      });
+      parsedRequest.queryParams = parsedQueryParams;
+
+      const parsedHeaders = sanitizedHeaders.map((headerKeyValue) => {
+        return {
+          id: headerKeyValue.id,
+          key: renderVariables(headerKeyValue.key),
+          value: renderVariables(headerKeyValue.value),
+        };
+      });
+      parsedRequest.headers = parsedHeaders;
+
+      if (request.body) {
+        const parsedBody =
```

Comment:
Instead of all this logic, can we create a common render method in EnvVariableService which recursively renders a JSON object?